src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -153,7 +167,10 @@ Status CoreWorkerDirectActorTaskSubmitter::PushTask(rpc::DirectActorClient &clie
               store_provider_->Put(RayObject(data_buffer, metadata_buffer), object_id));
         }
       });
-  return status;
+  if (!status.ok()) {
+    TreatTaskAsFailed(task_id, num_returns, rpc::ErrorType::ACTOR_DIED);
+  }
+  return Status::OK();
```

Comment:
Should we be returning `OK()` here? If so, make sure it's clearly documented - not the behavior I would expect without context.

Reply:
I'll change this to return void and update comment.